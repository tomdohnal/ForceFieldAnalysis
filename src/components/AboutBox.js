// @flow
import React from 'react';
import { Divider } from 'semantic-ui-react';

import { Box, Text } from './common';
import { COLORS } from '../constants/styles';

type Props = {
  onCloseLinkClick: ()=>void,
}

const AboutBox = ({
  onCloseLinkClick,
}: Props) => (
  <Box
    position="absolute"
    zIndex="999"
    backgroundColor={COLORS.WHITE}
    width="500px"
    right="30px"
    padding="16px"
    textAlign="initial"
    border={`2px ${COLORS.BLUE} solid`}
  >
    <Text
      onClick={onCloseLinkClick}
      position="absolute"
      right="16px"
      top="8px"
      clickable
    >Close
    </Text>
    <Text fontSize="32px" marginTop="16px">What is Force Field Analysis?</Text>
    <Divider hidden />
    <Text fontSize="18px">Force Field Analysis is a decision-making tool. Originally created in 1940s by psychologist Kurt Lewin, it is widely used in business for making decisions. The idea is to create driving (also called helping) forces that are driving change and hindering (also called resisting) forces that are hindering change. You can attribute different strengths to each force. Once you&#39;ve done with that, you can compare the sum of your driving forces and your hindering forces and decide if you should go or not go for change!</Text>
  </Box>
);

export default AboutBox;
