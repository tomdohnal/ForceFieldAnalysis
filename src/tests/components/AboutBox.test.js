import React from 'react';
import { shallow } from 'enzyme';

import AboutBox from '../../components/AboutBox';

describe('AboutBox', () => {
  const onCloseLinkClick = jest.fn();
  const ShallowAboutBox = shallow(<AboutBox onCloseLinkClick={onCloseLinkClick} />);

  it('renders correctly', () => {
    expect(ShallowAboutBox).toMatchSnapshot();
  });

  it('triggers onCloseLinkClick after the link is clicked', () => {
    ShallowAboutBox.find('[clickable=true]').simulate('click');

    expect(onCloseLinkClick).toBeCalled();
  });
});
