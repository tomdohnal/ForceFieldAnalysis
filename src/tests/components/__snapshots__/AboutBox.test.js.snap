// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutBox renders correctly 1`] = `
<boxStyles
  backgroundColor="#fff"
  border="2px #268bd2 solid"
  padding="16px"
  position="absolute"
  right="30px"
  textAlign="initial"
  width="500px"
  zIndex="999"
>
  <text
    clickable={true}
    onClick={[Function]}
    position="absolute"
    right="16px"
    top="8px"
  >
    Close
  </text>
  <text
    fontSize="32px"
    marginTop="16px"
  >
    What is Force Field Analysis?
  </text>
  <Divider
    hidden={true}
  />
  <text
    fontSize="18px"
  >
    Force Field Analysis is a decision-making tool. Originally created in 1940s by psychologist Kurt Lewin, it is widely used in business for making decisions. The idea is to create driving (also called helping) forces that are driving change and hindering (also called resisting) forces that are hindering change. You can attribute different strengths to each force. Once you've done with that, you can compare the sum of your driving forces and your hindering forces and decide if you should go or not go for change!
  </text>
</boxStyles>
`;
